### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by awendt**
Have a look at the [rendered README](https://github.com/babbel/terraform-aws-s3-bucket-with-cross-region-replication/blob/68ef11506152af06b65505722f49eda39606ff10/README.md)

### CODE REVIEW

**Review by jtsaito — APPROVED** [review]

File: primary.tf [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  provider = aws.primary
+
+  bucket = var.primary_name
+
+  force_destroy = var.force_destroy
+
+  tags = var.tags
+}
+
+resource "aws_s3_bucket_versioning" "primary" {
```
Comment (jansiwy) [inline primary.tf:11]: I'd recommend to mention explicitly in the README that it's not possible to set `aws_s3_bucket_versioning` on the created buckets outside of the module, as the module's versioning configuration is required for the replication to work.

File: primary.tf [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  depends_on = [aws_s3_bucket_versioning.primary]
+
+  role   = aws_iam_role.replication.arn
+  bucket = aws_s3_bucket.primary.bucket
+
+
+  rule {
+    id = aws_s3_bucket.secondary.bucket
+
+    filter {}
```
Comment (jansiwy) [inline primary.tf:33]: let's mention in a comment what this empty block means.

File: secondary.tf [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  provider = aws.secondary
+
+  bucket = aws_s3_bucket.secondary.bucket
+
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket_acl" "secondary" {
```
Comment (jansiwy) [inline secondary.tf:21]: `provider` is missing... however, see also my comment that I'd prefer to omit this resource in a public module.

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  type = string
+
+  description = <<EOS
+Name of bucket in created via the `aws.secondary` AWS provider.
+EOS
+}
+
+variable "tags" {
+  type    = map(string)
+  default = {} # Use structural Object Type in Terraform 0.12
```
Comment (jansiwy) [inline variables.tf:30]: the comment makes no sense:

```suggestion
  default = {}
```

**Review by jansiwy — APPROVED** [review]
apart from that: LGTM!

File: primary.tf (hunk 1 above)
Comment (awendt) [reply primary.tf:11]: see 36c7e6283f4f00471fb4213f29341d1c659b66df

File: primary.tf (hunk 2 above)
Comment (awendt) [reply primary.tf]: see 40c93749c9dbd7fd158bdd06f0740b5aaee5bc33

**Pull request merged**